apis/oauth/src/main/java/org/jclouds/oauth/v2/config/OAuthProperties.java

```diff
@@ -36,6 +36,35 @@
     */
    public static final String CREDENTIAL_TYPE = "jclouds.oauth.credential-type";
 
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, a "resource" must
+    * be specified as part of the request.
+    *
+    * @see <a href="https://msdn.microsoft.com/en-us/library/azure/dn645543.aspx">doc</a>
+    */
+   public static final String RESOURCE = "jclouds.oauth.resource";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, an "azure-credential"
+    * is a password.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_CREDENTIAL = "oauth.azure-credential";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, an "azure-identity"
+    * is the GUID associated with an application given permssion to perform operations in
+    * Azure Resource Manager.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_IDENTITY = "oauth.azure-identity";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, the azure-endpoint
+    * is the URL https://login.microsoftonline.com/<Tenant ID>/oauth2/token, where <Tenant ID>
+    * corresponds to your Azure AD Tenant.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_ENDPOINT = "oauth.azure-endpoint";
```

Comment:
LiveTest could just "authenticateTest" and switch based off credential type and asset Token present after switch.  Seems cleanest to me.